Cannot compile Luau.Repl.CLI on Android device from Termux

When compiling the Luau CLI (Luau.Repl.CLI) from a Android device via Termux, the following error below will appear:
```
luau/CLI/FileUtils.cpp:156:24: error: use of undeclared identifier 'IFTODT'
                type = IFTODT(st.st_mode);
                       ^
1 error generated.
make[3]: *** [CMakeFiles/Luau.Repl.CLI.dir/build.make:76: CMakeFiles/Luau.Repl.CLI.dir/CLI/FileUtils.cpp.o] Error 1
make[2]: *** [CMakeFiles/Makefile2:261: CMakeFiles/Luau.Repl.CLI.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:268: CMakeFiles/Luau.Repl.CLI.dir/rule] Error 2
make: *** [Makefile:202: Luau.Repl.CLI] Error 2
```
![Screenshot_20211104-235510_Termux](https://user-images.githubusercontent.com/30568955/140384070-47ba5562-da60-49d3-8766-c5bb5c67f497.jpg)


Notes:
- Luau CLI compiles fine on my laptop (running at x86_64 arch, tested at Linux)
- The Luau AST, VM, and Compiler libraries can compile normally (only the CLI has this issue)

Device used to compile:
Samsung Galaxy S20 FE SM-G780G
8 GB of RAM
Snapdragon 865 (ARM64)
Android 11 One UI 3.1 (Android security patch level October 1 2021)

Commands used:
```
git clone https://github.com/Roblox/luau.git
cd luau && mkdir build && cd build
cmake .. -DCMAKE_BUILD_TYPE=RelWithDebInfo
cmake --build . --target Luau.Repl.CLI
```

EDIT 1: Added more information
EDIT 2: Typo fix

--- a/CLI/FileUtils.cpp
+++ b/CLI/FileUtils.cpp
@@ -142,6 +142,7 @@ static bool traverseDirectoryRec(const std::string& path, const std::function<vo
             joinPaths(buf, path.c_str(), data.d_name);
 
             int type = data.d_type;
+            int mode = -1;
 
             // we need to stat DT_UNKNOWN to be able to tell the type
             if (type == DT_UNKNOWN)
@@ -153,18 +154,18 @@ static bool traverseDirectoryRec(const std::string& path, const std::function<vo
                 lstat(buf.c_str(), &st);
 #endif
 
-                type = IFTODT(st.st_mode);
+                mode = st.st_mode;
             }
 
-            if (type == DT_DIR)
+            if (type == DT_DIR || mode == S_IFDIR)
             {
                 traverseDirectoryRec(buf, callback);
             }
-            else if (type == DT_REG)
+            else if (type == DT_REG || mode == S_IFREG)
             {
                 callback(buf);
             }
-            else if (type == DT_LNK)
+            else if (type == DT_LNK || mode == S_IFLNK)
             {
                 // Skip symbolic links to avoid handling cycles
             }

